Fix Babel/Storybook Setup

sareh opened the pull request:
Resolves #115

Storybook should work and all components should load in styles.  Note: Reith font and normalise setup are not yet added. Those will be added in a separate PR.

**Testing instructions**

```
git checkout fix-babel-storybook-setup
npm install 
```
This installs initial dependencies & all individual packages
```
npm run storybook 
```
This runs `clear_styled_components` & then starts storybook

Visit http://localhost:8080  

There should be no errors, and styles should be pulled in.

- [ ] Tests added for new features
- [ ] Test engineer approval

radiocontrolled committed:
WIP

bcmn commented:
So there's an interesting behaviour with this commit:
`npm run storybook` fails if `.babelrc` isn't present (with the `exports read-only` error)
`npm run build` fails if `babel.config.js` isn't present

[It looks like Storybook simply doesn't support `babel.config.js` for the time being](https://storybook.js.org/configurations/custom-babel-config/), so I guess we're stuck with working out how to make `npm run build` use the `.babelrc` file. Looking into this is my next move this morning.

ChrisBAshton commented:
Can we create a `.babelrc` in the `.storybook` dir which has this:

```
{ "extends": "../babel.config.js" }
```

...?

ChrisBAshton commented on package.json:
```diff
@@ -4,7 +4,7 @@
   "main": "index.js",
   "scripts": {
-    "build": "lerna exec --parallel -- babel src -d dist --ignore test.jsx,stories.jsx",
+    "build": "lerna exec --parallel -- babel src -d dist --ignore ‘**/*.test.jsx,**/*.stories.jsx’ --root-mode upward",
```
These are non-ASCII quote marks 🤔 

```suggestion
    "build": "lerna exec --parallel -- babel src -d dist --ignore '**/*.test.jsx,**/*.stories.jsx' --root-mode upward",
```

radiocontrolled replied:
```suggestion
    "build": "lerna exec --parallel -- babel src -d dist --ignore '**/*.test.jsx,**/*.stories.jsx' --root-mode upward",
```

ChrisBAshton commented on babel.config.js:
```diff
@@ -0,0 +1,9 @@
+  env: {
+    test: {
+      presets: ['@babel/preset-env', '@babel/preset-react'],
+      plugins: ['@babel/plugin-transform-modules-commonjs'],
```
Doesn't this error? We don't have it in our dependencies!

ChrisBAshton replied:
Apparently not! Must be included in core.

ChrisBAshton replied:
Just re-installed and am now seeing the error again.

ChrisBAshton committed:
Merge branch 'latest' into fix-babel-storybook-setup

ChrisBAshton commented on package.json:
```diff
@@ -43,7 +43,6 @@
     "@storybook/addon-options": "^3.4.11",
     "@storybook/addon-viewport": "^3.4.11",
     "@storybook/react": "4.0.0-alpha.8",
-    "babel-core": "^7.0.0-bridge.0",
```
Just tried locally and this doesn't appear to be needed - perhaps because we have that plugin-transform-modules-commonjs plugin for Jest now 🤔

ChrisBAshton replied:
Nope - wait - after running `npm ci` and then `npm test`, it fails "cannot find babel-core" so we do need it.

radiocontrolled replied:
What is the reason for using the alpha of @storybook/react as opposed to something more recent - `"^4.0.12"`?

ChrisBAshton committed:
re-adds babel-core version for 'npm test' jest dependency

ChrisBAshton committed:
got Storybook compiling

ChrisBAshton committed:
fixes Storybook styles by ensuring styled-components singleton

After much investigation, it looks as though the Storybook instance is overwritten by Storybook instances inside node_modules folders under the packages/components/** directory. This is our short-term solution; we intend to raise an issue with Storybook.

sareh committed:
Update package-lock

bcmn reviewed (approved):
Really good work hunting down all these issues, all!

![mission_accomplie](https://media.giphy.com/media/lfd2RbUVygdiSHIUdp/giphy-downsized.gif)

ChrisBAshton committed:
Merge branch 'latest' into fix-babel-storybook-setup

sareh committed:
Merge branch 'latest' into fix-babel-storybook-setup
